### PR TITLE
Updated corner cases and Abi

### PR DESCRIPTION
**Pull request by akmahaja**

### CODE REVIEW

File: contracts/Freelancer.sol [hunk 1]
```diff
@@ -128,6 +130,10 @@ contract Freelancer {
         } else if (agreement.client.vote == Vote.declined && agreement.freelancer.vote == Vote.declined) {
             agreement.client.addr.transfer(agreement.value);
             emit transferFunds();
+        } else if (agreement.freelancer.vote == Vote.undecided) {
+            emit TriggerVote("Freelancer", agreement);
```
Comment (dsinghvi) [inline contracts/Freelancer.sol]: what is this supposed to do?

File: contracts/Freelancer.sol [hunk 2]
```diff
@@ -128,6 +130,10 @@ contract Freelancer {
...
+        } else {
+            emit Resolution(agreement);
```
Comment (dsinghvi) [inline contracts/Freelancer.sol]: in the else case we dont have resolution right?

Comment (dsinghvi) [reply contracts/Freelancer.sol]: or what about the case where one has voted and the other hasnt yet ? do we want to emit this event then?

Comment (akmahaja) [reply contracts/Freelancer.sol]: For the second one the `TriggerVote` event does exactly that.

Comment (akmahaja) [reply contracts/Freelancer.sol]: >in the else case we dont have resolution right?

We have covered all cases above, when both agree, both disagree or one votes and other undecided. The only thing left is both disagree, that's when trigger the `Resolution` event

**Pull request merged**